### ISSUE TITLE
MongoDB: 2 queries for findUnique lead to error

### ISSUE THREAD

**Issue by blazestudios23** [prisma/prisma#8276]
### Bug description

When two queries come in at the same time that execute a prisma.findUnque on our Mongo DB we get the following error:

```
[7/13/2021] [1:39:22 PM] [auth.ts] › ✖  error     Error: Error occurred during query execution: 
ConnectorError(ConnectorError { user_facing_error: Some(KnownError { message: "Inconsistent column data: Failed to convert 'List([String(\"60db4f12fcb8ff3c591c52eb\"), String(\"60db4f12fcb8ff3c591c52eb\")])' to 'ObjectId'.", meta: Object({"message": String("Failed to convert 'List([String(\"60db4f12fcb8ff3c591c52eb\"), String(\"60db4f12fcb8ff3c591c52eb\")])' to 'ObjectId'.")}), error_code: "P2023" }), kind: ConversionError(Failed to convert 'List([String("60db4f12fcb8ff3c591c52eb"), String("60db4f12fcb8ff3c591c52eb")])' to 'ObjectId'.) })
    at /graphql-gateway/node_modules/@prisma/client/runtime/index.js:28913:19
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

### How to reproduce

We have a front end app that is sending two different queries to the backend, the backend firs gets the user with a findUnqiue query. the queries are being pooled. We were able to get it working by using findFirst. 

### Expected behavior

Prisma should be able to find a unique value. 

### Prisma information

```
datasource db {
  provider = "mongodb"
  url      = env("MONGODB_URI")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["mongodb"]
}

model DbUser {
   id    String @id @default(dbgenerated()) @map("_id") @db.ObjectId
   createdAt DateTime @default(now())
   updatedAt DateTime @updatedAt
  firstName String?
  lastName String?
  email String?
  image String?
  phone String?
  count Int?
  role String

  @@map("dbUsers")
}

model User {
  id    String    @id @default(dbgenerated()) @map("_id") @db.ObjectId
  username String?
  email String    @unique
  redirect String?
  name String?
  locale String?
  image String?
  emailVerified Boolean

  @@map("users")
}
```

```
const dashboardUser = await prisma.dbUser.findFirst({
        where: { id: decodedToken.Id },
      });
```

### Environment & setup

- OS: Linux 
- Database: MongoDB
- Node.js version: 12.22.1

### Prisma Version

```
2.27.0
```

**Comment by matthewmueller** [comment]
Just to confirm, you're running `findUnique` on the `id`, not on the `email`? 

Right now we don't support managing @unique indexes, so you have to create them outside of Prisma. See [this issue](https://github.com/prisma/prisma/issues/6727) for more context.

**Comment by blazestudios23** [comment]
Yes, on the ID of the table.

**Comment by dpetrick** [comment]
Fixed in 2.28.
